core/src/main/java/org/apache/iceberg/actions/BaseRewriteTablePath.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import org.immutables.value.Value;
+
+@Value.Enclosing
+@SuppressWarnings("ImmutablesStyle")
+@Value.Style(
+    typeImmutableEnclosing = "ImmutableRewriteTablePath",
+    visibilityString = "PUBLIC",
+    builderVisibilityString = "PUBLIC")
+interface BaseRewriteTablePath extends RewriteTablePath {
```

Comment:
Question: why do we need this interface?

Reply:
It's needed for the [results of the action](https://github.com/apache/iceberg/blob/154e8ddb51709b45f3d6ec8310e124ed8f90cfbb/api/src/main/java/org/apache/iceberg/actions/RewriteTablePath.java#L77-L91) and the following the same immutable implementation standard [followed in other actions](https://github.com/apache/iceberg/pull/10920#discussion_r1755114815)